src/background/StreamSessionManager.ts

```diff
@@ -0,0 +1,373 @@
+            // If the number of users changed, notify the current user
+            const delta = numUsersAfter - numUsersBefore;
+
+            this.notifyUserCountChanges(delta, false);
+        }
+
+        this.sessionState = {
+            leaderId: message.state.leaderId,
+            sessionJoinOrder: message.state.sessionJoinOrder,
+            sessionUsers: this.sessionState.sessionUsers
+        };
+
+        console.warn(this.sessionState);
+
+        if (this.connectionState === ConnectionState.ConnectingToPeer) {
+            this.connectToSession();
+        }
+    }
+
+    /**
+     * Connects the current user to the session
+     */
+    private connectToSession(): void {
+        console.warn(`Connecting to session. ReferralId: ${this.connectToExistingSessionReferralId}`);
+        if (this.connectionState !== ConnectionState.ConnectingToPeer) {
+            // This method should only be called once we've received state info from our contact
+            // This indicates a bug in our code
+            console.error(`ConnectToSession called with invalid connection state: ${this.connectionState}`);
+        }
+
+        this.connectionState = ConnectionState.ConnectingToSession;
+
+        this.sessionState.sessionUsers.forEach(user => {
+            if (user.id !== this.connectToExistingSessionReferralId && user.id !== this.currentUserId) {
+                this.connectToPeer(user.id);
+            }
+        });
+
+        this.notifyUserCountChanges(this.sessionState.sessionUsers.size, true);
+
+        // Wait for new connections to form so the topology is set up correctly then request the latest state from leader
+        setTimeout(() => {
+            const request: IStateSyncRequestMessage = { messageType: MessageType.StateSyncRequest };
+            const leader = this.sessionState.sessionUsers.get(this.sessionState.leaderId);
+            console.warn("got leader");
+            console.warn(leader);
+            leader?.connection?.send(request);
+
+            this.connectionState = ConnectionState.Connected;
+        }, 500);
+    }
+
+    /**
+     * Notifies subscribers of a change in the session users list
+     * @param delta The change in amount of users in the party
+     */
+    private notifyUserCountChanges(delta: number, isSessionJoin: boolean): void {
+        this.onUserCountChange(this.sessionState.sessionUsers.size - 1, delta, isSessionJoin);
+    }
+
+    /**
+     * Method to handle when a peer connects to the current user if the current user is the host
+     * @param conn The peer connection
+     */
+    private onNewConnection(conn: Peer.DataConnection): void {
+        console.warn("got new connection");
+        console.warn(conn);
+        // Push newly connected user to the end of the session join order
+        this.sessionState.sessionJoinOrder.push(conn.peer);
+
+        const connectionMetadata: IConnectToPeerMetadata = conn.metadata as IConnectToPeerMetadata;
```

Comment:
You don't need both `: IType` and `as IType`.

Reply:
Copy pasta :P will fix